nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MonitorActivity.java

```diff
@@ -217,224 +230,221 @@ protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        } else {
+            context.yield();    // no need to dominate CPU checking times; let other processors run for a bit.
+        }
+
+        if (wasActive && !canBecomeInactive) {
+            // We need to block ACTIVE -> INACTIVE state transition, because we are not connected to the cluster.
+            // When we reconnect, and the INACTIVE state persists, then the next onTrigger will do the transition.
+            logger.trace("ACTIVE->INACTIVE transition is blocked, because we are not connected to the cluster.");
+        } else {
+            this.wasActive.set(isActive);
+            this.inactivityStartMillis.set(lastActivity);
+        }
+    }
+
+    protected long getStartupTime() {
+        return System.currentTimeMillis();
+    }
+
+    protected final long getLatestSuccessTransfer() {
+        return localFlowActivityInfo.getLastSuccessfulTransfer();
+    }
+
+    private String tryLoadLastSuccessfulTransfer(ProcessContext context) {
+        final StateManager stateManager = context.getStateManager();
+        try {
+            final StateMap localStateMap = stateManager.getState(Scope.LOCAL);
+            return localStateMap.get(STATE_KEY_LOCAL_FLOW_ACTIVITY_INFO);
+        } catch (IOException e) {
+            throw new ProcessException("Failed to load local state due to " + e, e);
+        }
+    }
+
+    private void synchronizeState(ProcessContext context) {
+        final ComponentLog logger = getLogger();
+        final boolean isConnectedToCluster = context.isConnectedToCluster();
+
+        if (isReconnectedToCluster(isConnectedToCluster)) {
+            localFlowActivityInfo.forceSync();
+            connectedWhenLastTriggered.set(true);
+        }
+        if (!isConnectedToCluster) {
+            connectedWhenLastTriggered.set(false);
+        } else if (localFlowActivityInfo.syncNeeded()) {
+            final CommonFlowActivityInfo commonFlowActivityInfo = new CommonFlowActivityInfo(context);
+            localFlowActivityInfo.update(commonFlowActivityInfo);
+
+            try {
+                commonFlowActivityInfo.update(localFlowActivityInfo);
+                localFlowActivityInfo.setNextSyncMillis();
+            } catch (final SaveSharedFlowStateException ex) {
+                logger.debug("Failed to update common state.", ex);
             }
+        }
+    }
 
-            if (sendInactiveMarker && shouldThisNodeReport(isClusterScope, shouldReportOnlyOnPrimary, context)) {
-                lastInactiveMessage.set(System.currentTimeMillis());
+    private void onTriggerInactiveFlow(ProcessContext context, ProcessSession session, boolean isClusterScope, long lastActivity) {
+        final ComponentLog logger = getLogger();
+        final boolean shouldReportOnlyOnPrimary = shouldReportOnlyOnPrimary(isClusterScope, context);
+        final boolean shouldThisNodeReport = shouldThisNodeReport(isClusterScope, shouldReportOnlyOnPrimary, context);
```

Comment:
`shouldThisNodeReport()` could call `shouldReportOnlyOnPrimary()` internally because all the parameters are passed to that method too and `shouldReportOnlyOnPrimary` local variable is not used anywhere else.

Also in `onTriggerActiveFlow()`.